.gitignore

```diff
@@ -1,12 +1,26 @@
-*~
+# OS X
 .DS_Store
 
+# Others
+*~
 *.log
-/node_modules
-fixtures/**/node_modules
 
+# VSCode
 .vscode/
+tsconfig.json
 jsconfig.json
+
+# NodeJS
+node_modules/
```

Comment:
We sure fake `node_modules` in our tests: https://github.com/callstack/haul/blob/master/src/utils/__tests__/fixtures/node_modules/react-native/package.json
Maybe lets add something like `!__tests__/**/node_modules`

Reply:
I think that's not necessary, as `node_modules` in fixtures are untouched

Reply:
Just checked, not being ignored

Reply:
Yeah,`node_modules/` will ignore any directory matching this glob. we need to add `!__tests__/**/node_modules` if we don't wanna ignore it

Reply:
`git check-ignore src/utils/__tests__/fixtures/node_modules/react-native/package.json`

Reply:
You're wrong @satya164 

Using `git check-ignore` we can see that the `./src/utils/__tests__/fixtures/node_modules/react-native/package.json` __is not ignored__ by `node_modules/` rule.

```bash
git check-ignore ./src/utils/__tests__/fixtures/node_modules/react-native/package.json # doesn't output anything -> not ignored

git check-ignore ./node_modules/react-native/package.json # outputs path -> ignored
```
and also

```bash
git check-ignore ./src/utils/__tests__/fixtures/node_modules/react-native/package.json --no-index
echo $? # gives exit code 1 -> None of the provided paths are ignored.

git check-ignore ./node_modules/react-native/package.json --no-index
echo $? # gives exit code 0 -> One or more of the provided paths is ignored.
```

Reply:
If the file is already being tracked by git, it's not affected by entries in .gitignore

Reply:
@satya164 Fair enough. Should we still add it to ignore, as it's already been tracking?

Reply:
@satya164 That's why I used `--no-index` option, it doesn't care if the file was tracked or not - the result is still the same.

Reply:
@zamotany I dunno how you are testing, but if I remove `!**/__tests__/**/node_modules` from `.gitignore`, I get:

```sh
git check-ignore ./src/utils/__tests__/fixtures/node_modules/react-native/package.json --no-index # exit code is 0

git check-ignore ./src/utils/__tests__/fixtures/node_modules/react-native/package.json # exit code is 1
```

You can also test that the `.gitignore` file affects `./src/utils/__tests__/fixtures/node_modules` folder by creating a file under that folder and running `git status`, which shouldn't output any untracked files.

Reply:
> @Krizzu Should we still add it to ignore, as it's already been tracking?

Yeah, I think we should. It'll be confusing to have something being tracked and while mtaching the ignore filters.

Reply:
@satya164 if you look at latest commit, you'll see this already done, I've shown @Krizzu what's the issue locally :)

Reply:
@thymikee yeah, saw it. I was just trying to disprove this statement by @zamotany 😛

![image](https://user-images.githubusercontent.com/1174278/35106002-173d3d04-fc93-11e7-8915-daa6c41e5752.png)

Reply:
Let's just merge it :D